### PR TITLE
Added streamer mode to hide sensible informations

### PR DESCRIPTION
**Pull request by CalDrac**

### CODE REVIEW

File: HintMachine/Views/LoginWindow.xaml [hunk 1]
```diff
@@ -5,20 +5,25 @@
...
         mc:Ignorable="d"
         Title="HintMachine" Height="345" Width="466" Background="#FFEEEEEE" ResizeMode="CanMinimize" Icon="/assets/icon.png">
-    <Grid HorizontalAlignment="Center" Width="400" Height="160" VerticalAlignment="Top" Margin="0,138,0,0">
+    <Grid HorizontalAlignment="Center" Width="400" Height="171" VerticalAlignment="Top" Margin="30,138,30,0">
         <Image HorizontalAlignment="Left" Height="110" Margin="122,-132,0,0" VerticalAlignment="Top" Width="180" 
                 Source="/Assets/logo_small.png"/>
-        
-        <Label Content="Archipelago host:" HorizontalAlignment="Left" VerticalAlignment="Top"/>
+
+        <Label Content="Archipelago host :" HorizontalAlignment="Left" VerticalAlignment="Top"/>
```
Comment (Dinopony) [inline HintMachine/Views/LoginWindow.xaml]: In English, there is no space before a colon (:) character.

File: HintMachine/Views/MainWindow.xaml.cs [hunk 2]
```diff
@@ -85,7 +86,16 @@ protected void OnArchipelagoSessionChange()
...
+            
+            if (!Settings.StreamerMode)
+            {
+                Logger.Log($"Connected to Archipelago session at {HintMachineService.Host} as {HintMachineService.Slot}.", LogMessageType.CONNEXION);
+                ConnectionTextBlock.Visibility = Visibility.Visible;
+            }
+            else {
+                Logger.Info($"Connected to Archipelago session.");
+                ConnectionTextBlock.Visibility = Visibility.Hidden;
+            }
```
Comment (Dinopony) [inline HintMachine/Views/MainWindow.xaml.cs]: Here, you check if you are in StreamerMode a first time, and then it's checked a second time by using the LogMessageType.CONNEXION.
I feel like this is the right place where to check it, therefore making the new LogMessageType unnecessary.

File: HintMachine/Views/MessageLog.xaml.cs [hunk 3]
```diff
@@ -204,7 +205,9 @@ public static bool CanDisplayMessage(string message, LogMessageType logMessageTy
                 return false;
             if (logMessageType == LogMessageType.ITEM_SENT && !Settings.DisplayItemSentMessages)
                 return false;
-
+            if (logMessageType == LogMessageType.CONNEXION && Settings.StreamerMode) {
+                return false;
+            }
```
Comment (Dinopony) [inline HintMachine/Views/MessageLog.xaml.cs]: See comment in MainWindow.xaml.cs, this is the second place where you check if we should display the connection message.

**Review by Dinopony — CHANGES_REQUESTED** [review]

**Comment by CalDrac** [comment]
I commented the MessageLog.xaml.cs code you pointed out.

Connected without checking StreamerMode
Checked StreamerMode

image

I think the filter for CONNEXION in MessageLog.xaml.cs is still needed.

**Review by Dinopony — APPROVED** [review]

**Pull request merged**